CP1A - Office Space Allocation

asce1062 opened the pull request:
**What does this PR do?**

A room allocation system for one of Andela’s facilities called Amity.


**Description of Task to be completed?**

- Specifing different db name during save state.
- Clearing DB on load state case. 
- Prints statement alerts Red, Yellow, Green and their meanings.
- Try implementing self documenting comments.
- Relevant directories and files.
- Extra edge case: Delete rooms and people.


**How should this be manually tested?**

`nosetests -v`


#### Screenshots (if appropriate)

![amityville](https://i.imgur.com/ZVYoZet.png)

Commit:
Beautified Python Code

Commit:
Add tests for load and save states. Add load and save states functionality.

Commit:
Checkpoint. Update docopt help information. Call clear_db before save_state and after load_state

Prevent saving already existing data. Also prevent loading already existing data.

Commit:
remove unused import

Commit:
remove unused import

karendi commented on models/amity.py:
```diff
@@ -1,80 +1,136 @@
 
 from models.room import LivingSpace, Office
 from models.person import Fellow, Staff
+from databases.database_models import *
```
import the specific objects/functions you want to use

karendi commented on models/amity.py:
```diff
@@ -84,461 +140,698 @@ def add_person(self, person_name, job_description, wants_accommodation='N'):
                 else:
                     return 'Invalid accommodation input.'
             elif job_description.upper() == 'STAFF':
+
```
Leave spaces between the code for better comprehension

karendi commented on models/amity.py:
```diff
@@ -84,461 +140,698 @@ def add_person(self, person_name, job_description, wants_accommodation='N'):
                 random_office = random.choice(vacant_offices)
+
                 # Insert the fellow name to the randomly chosen room.
+
```
Allocate the person using their unique identifiers

karendi commented on models/amity.py:
```diff
@@ -84,461 +140,698 @@ def add_person(self, person_name, job_description, wants_accommodation='N'):
                 self.unallocated_office.append(person_name.upper())
                 return 'No vaccant office.'
 
     def reallocate_person(self, person_id, room_name):
```
You can avoid the nested loops and just have one if statement checking for all the conditions

Commit:
implement feedback based on code review

Specifing different db name during save state. Clearing DB on load state case. Prints statement alerts Red, Yellow, Green and their meanings. Relevant directories and files. Extra edge case: Delete people.

Commit:
write tests for the delete method

Commit:
write tests and functionality for the delete_room method. Also include it in the interactive interface

Commit:
Sort Imports

Commit:
Sort more Imports

Commit:
Implement self documenting comments.

Commit:
Update UML diagrams and update README
